lightning/src/ln/channelmanager.rs

```diff
@@ -760,12 +768,36 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 			pending_events: Mutex::new(Vec::new()),
 			total_consistency_lock: RwLock::new(()),
 
+			persistence_lock: Arc::new((Mutex::new(false), Condvar::new())),
+
 			keys_manager,
 
 			logger,
 		}
 	}
 
+	/// Used to signal to users when the ChannelManager needs to be persisted to disk. This function
+	/// blocks either for the given Duration or indefinitely until the ChannelManager has updates and
+	/// needs persistence. It returns a bool indicating whether persistence is necessary (which will
+	/// always be true if max_wait is None).
+	pub fn block_until_needs_persist(&self, max_wait: Option<Duration>) -> bool {
```

Comment:
It would be good to move this to a utility class, since `ChannelManager` is already pretty large.

Reply:
For example:

```rust
pub struct Notifier {
    lock: Arc<(Mutex<bool>, Condvar)>
}

impl Notifier {
    pub fn block_until_notified(&self, max_wait: Option<Duration>) -> bool { ... }
    pub fn notify(&self) { ... }
}
```

Reply:
nice! thanks